test/config/_mock_logger.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+/**
+ * @typedef {object} LogEntry
+ * @property {string} type
+ * @property {string} message
+ * @property {Array<any>} message
+ */
+
+/**
+ * SomeClass is an example class for my question.
+ * @class
+ * @constructor
+ * @public
+ */
+class MockLogger {
```

Comment:
There is already a `class CaptureLogger` that is very similar. Is the intent to share those two?